graphql/src/lib.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+mod handler;
+
+use actix_web::{
+    dev::Server,
+    error::{ErrorMethodNotAllowed, ErrorNotFound, ErrorUnsupportedMediaType},
+    http::{header::CONTENT_TYPE, Method},
+    web,
+    web::{Payload, Query, ServiceConfig},
+    App, FromRequest, HttpRequest, HttpResponse, HttpServer, Result,
+};
+use coordinator::module::{HandleGraphQlRequest, SessionId};
+pub use handler::handle_gql_query;
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+// TODO: replace with the real Client.
+pub struct Client {}
+
+impl Client {
+    fn allocate_session(&self, _height: Option<u64>) -> SessionId {
+        123
+    }
+}
+
+pub struct ServerData {
```

Comment:
How about ServerContext?
`Data` is too general and does not deliver any meaning.
`Context` is aloso a bit general, but we can know it contains some conifuration or other useful objects that is used in GraphQL Server.

If the data is only used in Server construction, we may name it ServerConfig.

Reply:
The name `Data` is to matched with `actix-web`'s `Data`.
`server_data: web::Data<Arc<ServerData>>,`